python/tests/res/job_queue/test_job_queue.py

```diff
@@ -77,70 +76,67 @@ def test_kill_jobs(self):
             job_queue = create_queue(never_ending_script)
 
             assert job_queue.queue_size == 10
-            assert job_queue.is_running()
+            assert job_queue.is_active()
 
-            threads = start_all(job_queue)
+            start_all(job_queue)
 
             # make sure never ending jobs are running
-            time.sleep(3.0)
             wait_until(
-                lambda: self.assertTrue(job_queue.is_running())
+                lambda: self.assertTrue(job_queue.is_active())
             )
 
             for job in job_queue.job_list:
                 job.stop()
 
             wait_until(
-                lambda: self.assertFalse(job_queue.is_running())
+                lambda: self.assertFalse(job_queue.is_active())
             )
             
             for job in job_queue.job_list:
                 assert job.status == JobStatusType.JOB_QUEUE_IS_KILLED
             
-            for t in threads:
-                t.join()
+            for job in job_queue.job_list:
+                job.thread.join()
```

Comment:
is this necessary now?

Reply:
it is at least a safe guard to ensure all spawned threads are finished